### PR TITLE
Add master and hotfix support

### PR DESCRIPTION
**Pull request by venkatd**

### CODE REVIEW

File: lib/code/branch.rb [hunk 1]
```diff
@@ -121,6 +131,10 @@ def private?
       !! name.match(/-local$/)
     end
 
+    def hotfix?
+      !! name.match(/^hotfix-/)
```
Comment (JoshSmith) [inline lib/code/branch.rb]: This looks for starts with `hotfix-`?

Comment (venkatd) [inline lib/code/branch.rb]: Yeah exactly

File: bin/code [hunk 2]
```diff
@@ -36,10 +36,15 @@ command :start do |c|
...
+command :publish, :pr do |c|
+  c.switch [:master], desc: 'Make the pull request based off the master branch'
   c.action do |global_options, options, args|
     message = args.join(' ')
-    $git.publish message
+
+    base_branch = nil
+    base_branch = Code::Branch.master if options[:master]
+
+    $git.publish(branch: Code::Branch.current, base: base_branch, message: message)
```
Comment (JoshSmith) [inline bin/code]: I don't really understand what this is doing.

**Comment by JoshSmith** [comment]
Looks good in general. I think pretty idiomatic. Only things that's not clear to me is what's going on in `bin/code`.

File: bin/code (hunk 2 above)
Comment (venkatd) [inline bin/code]: The `branch` means the branch to create a PR for and the `base` is the one we want to merge it into. `$git` is an instance of the `Code::Git` object I wrote which wraps all the git commands we need with ruby methods.

File: bin/code [hunk 3]
```diff
@@ -36,10 +36,15 @@ command :start do |c|
 end
 
 desc 'Upload your feature branch to GitHub'
-command :publish do |c|
+command :publish, :pr do |c|
+  c.switch [:master], desc: 'Make the pull request based off the master branch'
```
Comment (idelahoz) [inline bin/code]: how is this conditional to the hotfix prefix ?

File: bin/code (hunk 2 above)
Comment (venkatd) [inline bin/code]: So for example, if we might have `branch = add-product-photos` and `base = development`. Or we might have `branch = hotfix-foo-bar and base=master`.

**Comment by venkatd** [comment]
`bin/code` is the syntax that is defined in the [gli gem](https://github.com/davetron5000/gli). They let you define command line commands kind of like you define rake tasks.

File: lib/code/git.rb [hunk 4]
```diff
@@ -69,23 +75,24 @@ def commit(message)
       System.call "commit -m \"#{message}\""
     end
 
-    def publish(message = '')
+    def publish(branch:, base: nil, message: '')
       raise NotOnFeatureBranchError, 'Must be on a feature branch to publish your code' unless current_branch.feature?
       raise UncommittedChangesError, 'You have uncommitted changes' if uncommitted_changes?
 
-      current_branch.push
+      base = infer_base_branch(branch) if base.nil?
```
Comment (idelahoz) [inline lib/code/git.rb]: ok I get it know.

**Pull request merged**